### PR TITLE
Rework api

### PR DESCRIPTION
**Pull request by AlpriElse**
# General comments
- Utilize whitespace to make code more readable.
- SQL statements should be formatted to be readable
- Better variable names are needed
- Separate into MVC (Database operations should be abstracted)

cc/ @Shimura233

### CODE REVIEW

File: api/routes/articles.js [hunk 1]
```diff
@@ -8,179 +8,208 @@ const pool = new Pool({
   ssl: true
 });
 
+const ArticleModel = require('../models/ArticleModel')
+
 //  GET /articles
-router.get('/',async (req, res) => {
-  const queryObject = url.parse(req.url,true).query;
```
Comment (AlpriElse) [inline api/routes/articles.js:13]: You do not need the `url` dependency. Read the ExpressJS documentation: https://expressjs.com/en/api.html#req.query

File: api/routes/articles.js [hunk 2]
```diff
@@ -8,179 +8,208 @@ const pool = new Pool({
...
+router.get('/', async (req, res) => {
+  const queryObject = url.parse(req.url, true).query;
   const articleName = queryObject.articleName;
+
   try {
     const client = await pool.connect();
-    var query = `SELECT aid,fid,citation FROM articles where upper(title)=upper('${articleName}')`;
```
Comment (AlpriElse) [inline api/routes/articles.js:17]: Make SQL statements readable.

File: api/routes/articles.js [hunk 3]
```diff
@@ -8,179 +8,208 @@ const pool = new Pool({
...
-    var articleQueryObject = {authorInfo:[],fieldInfo:[]};
+
+    var articleQueryObject = {
+      authorInfo: [],
+      fieldInfo: []
+    };
+
     if (result && result.rowCount > 0) {
       const aid = result.rows[0].aid;
-      //console.log(typeof aid);
```
Comment (AlpriElse) [inline api/routes/articles.js:22]: Don't keep commented debugging statements in commits.

File: api/routes/articles.js [hunk 4]
```diff
@@ -8,179 +8,208 @@ const pool = new Pool({
...
   try {
-      const partialArticle = {title:null,authors:[],publishYear:null,url:null,citations:null,fieldID:null};
-      const keys = Object.keys(frontArticle);
-      keys.forEach(key => {
-        partialArticle[key] = frontArticle[key];
-      });
-      await client.query('BEGIN');
-      var query = `INSERT INTO articles (title,pub_year,url,citation,fid) values ('${partialArticle.title}',${partialArticle.publishYear},'${partialArticle.url}',${partialArticle.citations},'${partialArticle.fieldID}') RETURNING aid`;
-      const rinserted = await client.query(query);
-      const insertaid = rinsertid.rows[0].aid;
```
Comment (AlpriElse) [inline api/routes/articles.js:79]: Was this endpoint tested? There's a typo here. `rinserted` vs `rinsertid`

Comment (Shimura233) [reply api/routes/articles.js:79]: No. This has not been tested. Can I modify it right here?

**Pull request merged**